internal/auth0/action/resource_trigger_action.go

```diff
@@ -0,0 +1,194 @@
+package action
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/auth0/go-auth0"
+	"github.com/auth0/go-auth0/management"
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"github.com/auth0/terraform-provider-auth0/internal/config"
+)
+
+// NewTriggerActionResource will return a new auth0_trigger_action resource.
+func NewTriggerActionResource() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: createTriggerAction,
+		ReadContext:   readTriggerAction,
+		DeleteContext: deleteTriggerAction,
+		Importer: &schema.ResourceImporter{
+			StateContext: importTriggerAction,
+		},
+		Description: "With this resource, you can bind an action to a trigger. Once an action is created and deployed, it can be attached (i.e. bound) to a trigger so that it will be executed as part of a flow. The list of actions reflects the order in which they will be executed during the appropriate flow.\n\nOrdering of an action within a specific flow is not currently supported when using this resource; the action will get appended to the end of the flow. To precisely manage ordering, it is advised to either do so with the dashboard UI or with the `auth0_trigger_bindings` resource.",
```

Comment:
Apologies for having missed this the first time 😞 and maybe a bit of a nit but wdyt about suggesting `auth0_trigger_actions` as a resource to manage the full list of actions here instead of soon to be deprecated one?

Also I'm not too sure about this mention:
```
The list of actions reflects the order in which they will be executed during the appropriate flow.
```

As we don't have any actions list in this resource now, is there any way we could phrase it a bit better?

Lmk wdyt.